src/js_realm.hpp

```diff
@@ -570,6 +570,9 @@ bool RealmClass<T>::get_realm_config(ContextType ctx, size_t argc, const ValueTy
                 }
 
                 config.migration_function = [=](SharedRealm old_realm, SharedRealm realm, realm::Schema&) {
+                    // the migration function called early so the binding context might not be set
+                    realm->m_binding_context.reset(new RealmDelegate<T>(realm, Context<T>::get_global_context(ctx)));
```

Comment:
This needs to check if the binding context is already set. You can trigger a migration after opening a realm (via _updateSchema), and recreating the binding context will break things like listeners on the realm.